pydantic_ai_slim/pydantic_ai/toolsets/wrapper.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+
+import inspect
+from dataclasses import dataclass
+
+from .._run_context import AgentDepsT, RunContext
+from ._abstract import AbstractToolset
+from ._run import RunToolset
+from ._wrapper import AbstractWrapperToolset
+
+
+@dataclass
+class WrapperToolset(AbstractWrapperToolset[AgentDepsT]):
+    """A toolset that wraps another toolset and delegates to it."""
+
+    async def rewrap_for_run(
+        self, wrapped: AbstractToolset[AgentDepsT], ctx: RunContext[AgentDepsT]
+    ) -> WrapperToolset[AgentDepsT]:
+        """Return a new instance that wraps the provided stand-in for the original wrapped toolset that's been prepared for a specific run step.
```

Comment:
Can you explain why this is necessary? You don't need to update the docstring I just feel like a text description might help me understand why this is necessary...